autocorpus/__main__.py

```diff
@@ -204,14 +208,19 @@ def main():
             base_dir = file_path.parent if not file_path.is_dir() else file_path
             try:
                 ac = Autocorpus(
-                    config,
                     base_dir=str(base_dir),
                     main_text=structure[key]["main_text"],
                     linked_tables=sorted(structure[key]["linked_tables"]),
                     table_images=sorted(structure[key]["table_images"]),
                     trained_data=trained_data,
                 )
-
+                if args.config:
+                    ac.config = ac.read_config(config)
+                elif args.default_config:
+                    default_config = DefaultConfig.string_to_constant(
+                        args.default_config
+                    )
+                    ac.config = DefaultConfig.load_config(default_config)
```

Comment:
Personally I think it's cleaner to pass the `config` into the `Autocorpus` constructor rather than setting it after the fact. I'm guessing that's why you split out the actual processing to the `process_files()` method?

In any case, you need to call that too, otherwise the input files won't be processed when you run AC from the command line:


```suggestion
                    ac.config = DefaultConfig.load_config(default_config)
                ac.process_files()
```

The regression test didn't catch this because it's exercising a different part of the code, but without it you get errors like this:

```
❯ auto-corpus -c autocorpus/configs/config_pmc_pre_oct_2024.json -t output -f tests/data/PMC/
Pre-Oct-2024/PMC8885717.html
100%|█████| 1/1 [00:00<00:00, 2576.35it/s, file=PMC8885717*, linked_tables=0, table_images=0]
Auto-CORPus has completed processing with some errors. Please inspect the log file for further details.
```

Reply:
I'd also lean towards building the config and then passing it to the constructor. However I think this is something that will be changed by the upcoming refactor, so it might not be too important.

And yes, it looks like `process_files` isn't being called anywhere, so that needs to be called after `ac.config` is set